src/applications/travel-pay/redux/actions.js

```diff
@@ -135,3 +156,272 @@ export function submitMileageOnlyClaim(appointmentData) {
+  payload: data,
+});
+const submitComplexClaimFailure = error => ({
+  type: SUBMIT_COMPLEX_CLAIM_FAILURE,
+  error,
+});
+
+export function submitComplexClaim(claimId, claimData) {
+  return async dispatch => {
+    dispatch(submitComplexClaimStart());
+
+    try {
+      const options = {
+        method: 'PATCH',
+        body: JSON.stringify(claimData),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      };
+
+      const apptUrl = `${
+        environment.API_URL
+      }/travel_pay/v0/complex_claims/${claimId}/submit`;
+      const response = await apiRequest(apptUrl, options);
+      dispatch(submitComplexClaimSuccess(response));
+    } catch (error) {
+      dispatch(submitComplexClaimFailure(error));
+    }
+  };
+}
+
+// Fetching complex claim details (including expenses)
+const fetchComplexClaimDetailsStart = () => ({
+  type: FETCH_COMPLEX_CLAIM_DETAILS_STARTED,
+});
+const fetchComplexClaimDetailsSuccess = data => ({
+  type: FETCH_COMPLEX_CLAIM_DETAILS_SUCCESS,
+  payload: data,
+});
+const fetchComplexClaimDetailsFailure = error => ({
+  type: FETCH_COMPLEX_CLAIM_DETAILS_FAILURE,
+  error,
+});
+
+export function getComplexClaimDetails(claimId) {
+  return async dispatch => {
+    dispatch(fetchComplexClaimDetailsStart());
+
+    try {
+      const claimUrl = `${environment.API_URL}/travel_pay/v0/claims/${claimId}`;
+      const response = await apiRequest(claimUrl);
+      dispatch(fetchComplexClaimDetailsSuccess(response));
+      return response;
+    } catch (error) {
+      dispatch(fetchComplexClaimDetailsFailure(error));
+      throw error;
+    }
+  };
+}
+
+// Updating an expense
+const updateExpenseStart = expenseId => ({
+  type: UPDATE_EXPENSE_STARTED,
+  expenseId,
+});
+const updateExpenseSuccess = data => ({
+  type: UPDATE_EXPENSE_SUCCESS,
+  payload: data,
+});
+const updateExpenseFailure = (error, expenseId) => ({
+  type: UPDATE_EXPENSE_FAILURE,
+  error,
+  expenseId,
+});
+
+export function updateExpense(claimId, expenseType, expenseId, expenseData) {
+  return async dispatch => {
+    dispatch(updateExpenseStart(expenseId));
+
+    try {
+      if (!expenseType) {
+        throw new Error('Missing expense type');
+      } else if (!expenseId) {
+        throw new Error('Missing expense id');
+      }
+
+      const options = {
+        method: 'PATCH',
+        body: JSON.stringify(expenseData),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      };
+
+      const expenseUrl = `${
+        environment.API_URL
+      }/travel_pay/v0/expenses/${expenseType}/${expenseId}`;
+      await apiRequest(expenseUrl, options);
+      const result = { ...expenseData, id: expenseId };
+
+      // Fetch the complete complex claim details and load expenses into store
+      try {
+        await dispatch(getComplexClaimDetails(claimId));
+      } catch (fetchError) {
+        // Silently continue if fetching details fails
```

Comment:
Why silently continue?

Reply:
I understand "silently" may be triggering 😄